src/terminal/kitty.rs

```diff
@@ -232,15 +232,20 @@ impl KittyImages {
 
         let (image_width, image_height) = image.dimensions();
 
-        let image = if image_width > terminal_size.width || image_height > terminal_size.height {
-            image.resize_to_fill(
-                terminal_size.width,
-                terminal_size.height,
-                FilterType::Nearest,
-            )
-        } else {
-            image
-        };
+        let (image, image_width, image_height) =
```

Comment:
I'm not sure I like this way of rebinding.  I think we ought to remove `image_width` and `image_height` entirely, and then, in the line below, compare `image.width()` and `image.height()`, and eventually use both again, which would also fix my other remark.

Reply:
Agreed, I had the same feeling about re-binding of the size values. Will rework the code.